x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -585,29 +586,36 @@ class AgentPolicyService {
       }
     }
 
-    const agentPolicies = await pMap(
-      agentPoliciesSO.saved_objects,
-      async (agentPolicySO) => {
-        const agentPolicy = mapAgentPolicySavedObjectToAgentPolicy(agentPolicySO);
-        if (withPackagePolicies) {
-          agentPolicy.package_policies =
-            (await packagePolicyService.findAllForAgentPolicy(soClient, agentPolicySO.id)) || [];
-        }
-        if (options.withAgentCount) {
-          await getAgentsByKuery(appContextService.getInternalUserESClient(), soClient, {
-            showInactive: true,
-            perPage: 0,
-            page: 1,
-            kuery: `${AGENTS_PREFIX}.policy_id:${agentPolicy.id}`,
-          }).then(({ total }) => (agentPolicy.agents = total));
-        } else {
-          agentPolicy.agents = 0;
-        }
+    const agentPolicies = agentPoliciesSO.saved_objects.map((agentPolicySO) => {
+      const agentPolicy = mapAgentPolicySavedObjectToAgentPolicy(agentPolicySO);
+      agentPolicy.agents = 0;
+      return agentPolicy;
+    });
 
-        return agentPolicy;
-      },
-      { concurrency: 50 }
-    );
+    if (options.withAgentCount || withPackagePolicies) {
```

Comment:
Not necessary related to that change, but this avoid a bottleneck with `p-map` when not needed, I could do a separate PR if you think it make more sense